habitat/plan.sh

```diff
@@ -64,58 +91,46 @@ do_prepare() {
 }
 
 do_build() {
-  local _libxml2_dir
-  local _libxslt_dir
-  local _zlib_dir
-  export CPPFLAGS
-  export GEM_HOME
-  export GEM_PATH
-  export NOKOGIRI_CONFIG
-
-  _libxml2_dir=$(pkg_path_for libxml2)
-  _libxslt_dir=$(pkg_path_for libxslt)
-  _zlib_dir=$(pkg_path_for zlib)
-
-  CPPFLAGS="${CPPFLAGS} ${CFLAGS}"
-  GEM_HOME=${pkg_prefix}/bundle
-  GEM_PATH=${GEM_HOME}
-  NOKOGIRI_CONFIG="--use-system-libraries \
-    --with-zlib-dir=${_zlib_dir} \
-    --with-xslt-dir=${_libxslt_dir} \
-    --with-xml2-include=${_libxml2_dir}/include/libxml2 \
-    --with-xml2-lib=${_libxml2_dir}/lib"
-
-  build_line "Executing bundle install inside hab-cache path. ($CACHE_PATH/chef-config)"
-  ( cd "$CACHE_PATH/chef-config" || exit_with "unable to enter hab-cache directory" 1
-    bundle config --local build.nokogiri "${NOKOGIRI_CONFIG}"
-    bundle config --local silence_root_warning 1
-    _bundle_install "${pkg_prefix}/bundle"
+  ( cd "$CACHE_PATH" || exit_with "unable to enter hab-cache directory" 1
+    build_line "Installing gem dependencies ..."
+    bundle install
+    build_line "Installing this project's gems ..."
+    bundle exec rake install
+    for gem in $GEM_HOME/bundler/gems/*; do
+      ( cd $gem
+        build_line "Installing gems from git repos properly ..."
+        rake install
+      )
+    done
   )
-
-  build_line "Executing bundle install inside source path. ($SRC_PATH)"
-  _bundle_install "${pkg_prefix}/bundle"
 }
 
 do_install() {
-  build_line "Copying directories from source to pkg_prefix"
-  mkdir -p "${pkg_prefix}/chef"
-  for dir in bin chef-bin chef-config chef-utils lib chef.gemspec Gemfile Gemfile.lock; do
-    cp -rv "${SRC_PATH}/${dir}" "${pkg_prefix}/chef/"
-  done
-
-  # If we generated them on install, bundler thinks our source is in $HAB_CACHE_SOURCE_PATH
-  build_line "Generating binstubs with the correct path"
-  ( cd "$pkg_prefix/chef" || exit_with "unable to enter pkg prefix directory" 1
-    _bundle_install \
-      "${pkg_prefix}/bundle" \
-      --local \
-      --quiet \
-      --binstubs "${pkg_prefix}/bin"
+  ( cd "$pkg_prefix" || exit_with "unable to enter pkg prefix directory" 1
+    export BUNDLE_GEMFILE="${CACHE_PATH}/Gemfile"
+    build_line "** fixing binstub shebangs"
+    fix_interpreter "${pkg_prefix}/vendor/bin/*" "$_chef_client_ruby" bin/ruby
+    export BUNDLE_GEMFILE="${CACHE_PATH}/Gemfile"
+    for gem in chef-bin chef inspec-core-bin ohai; do
+      build_line "** generating binstubs for $gem with precise version pins"
+      appbundler $CACHE_PATH $pkg_prefix/bin $gem
+    done
   )
+}
 
-  build_line "Fixing bin/ruby and bin/env interpreters"
-  fix_interpreter "${pkg_prefix}/bin/*" core/coreutils bin/env
-  fix_interpreter "${pkg_prefix}/bin/*" "$_chef_client_ruby" bin/ruby
+do_after() {
+  build_line "Trimming the fat ..."
```

Comment:
Very excited to see this. We probably need to come up with a unified way to do fat cleanup or at least move more of the cleanup we do in omnibus here. There's more fat to trim